src/test/java/org/folio/fqm/service/EntityTypeServiceTest.java

```diff
@@ -136,13 +156,87 @@ void shouldFilterBySearchText() {
   void shouldHandleNullSearchText() {
     UUID entityTypeId = UUID.randomUUID();
     String valueColumnName = "column_name";
+    EntityType entityType = new EntityType()
+      .id(entityTypeId.toString())
+      .name("yep")
+      .columns(List.of(new EntityTypeColumn().name(valueColumnName)));
     List<String> fields = List.of("id", valueColumnName);
     String expectedFql = "{\"" + valueColumnName + "\": {\"$regex\": " + "\"\"}}";
+    when(repo.getEntityTypeDefinition(entityTypeId)).thenReturn(Optional.of(entityType));
 
     entityTypeService.getColumnValues(entityTypeId, valueColumnName, null);
     verify(queryProcessorService).processQuery(entityTypeId, expectedFql, fields, null, 1000);
   }
 
+  @Test
+  void shouldReturnPredefinedValues() {
+    UUID entityTypeId = UUID.randomUUID();
+    String valueColumnName = "column_name";
+    List<ValueWithLabel> values = List.of(
+      new ValueWithLabel().value("value_01").label("value_01"),
+      new ValueWithLabel().value("value_02").label("value_02")
+    );
+    EntityType entityType = new EntityType()
+      .id(entityTypeId.toString())
+      .name("the entity type")
+      .columns(List.of(new EntityTypeColumn().name(valueColumnName).values(values)));
+
+    when(repo.getEntityTypeDefinition(entityTypeId)).thenReturn(Optional.of(entityType));
+
+    ColumnValues actualColumnValueLabel = entityTypeService.getColumnValues(entityTypeId, valueColumnName, "");
+
+    assertEquals(new ColumnValues().content(values), actualColumnValueLabel);
+  }
+
+  @Test
+  void shouldReturnValuesFromApi() {
+    UUID entityTypeId = UUID.randomUUID();
+    String valueColumnName = "column_name";
+    EntityType entityType = new EntityType()
+      .id(entityTypeId.toString())
+      .name("the entity type")
+      .columns(List.of(new EntityTypeColumn()
+        .name(valueColumnName)
+        .valueSourceApi(new ValueSourceApi()
+          .path("fake-path")
+          .valueJsonPath("$.what.ever.dude.*.theValue") // Approach 1: Explicitly use the full path
+          .labelJsonPath("$..theLabel") // Approach 2: Just dive all the way down and find everything with this key
```

Comment:
It's a little hidden, but I added the comments here to illustrate the different ways to do the JsonPath, since it seemed worth noting _somewhere_